src/daft-catalog/src/identifier.rs

```diff
@@ -33,58 +79,84 @@ impl Identifier {
             return Err(err);
         };
-        let Ok(parts) = parser.parse_multipart_identifier() else {
+        let Ok(idents) = parser.parse_multipart_identifier() else {
             return Err(err);
         };
-        if parts.is_empty() {
+        if idents.is_empty() {
             return Err(err);
         }
-        // TODO Identifier normalization is omitted until further discussion.
-        let mut parts = parts
-            .iter()
-            .map(|part| part.value.to_string())
-            .collect::<Vec<String>>();
-        let name = parts.pop().unwrap();
-        let namespace = parts;
+        let mut names = idents.into_iter().map(|i| i.into()).collect::<Vec<Name>>();
+        let name = names.pop().unwrap();
+        let namespace = names;
         Ok(Identifier::new(namespace, name))
     }
 }
 
+impl Display for Name {
+    /// Returns the ident as a string, adding double-quotes if delimited.
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if self.regular {
+            f.write_str(&self.text)
+        } else {
+            f.write_str(&format!("\"{}\"", &self.text))
+        }
+    }
+}
+
 impl Display for Identifier {
-    /// Joins the identifier to a dot-delimited path.
+    /// Returns the identifier as a dot-delimited string.
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         if self.namespace.is_empty() {
-            f.write_str(&self.name)
+            write!(f, "{}", &self.name)
         } else {
-            let prefix = self.namespace.join(".");
-            let string = format!("{}.{}", prefix, self.name);
-            f.write_str(&string)
+            let namespace = self
+                .namespace
+                .iter()
+                .map(|n| n.to_string())
+                .collect::<Vec<String>>()
+                .join(".");
+            let name = self.name.to_string();
+            write!(f, "{}.{}", namespace, name)
         }
     }
 }
 
-impl IntoIterator for Identifier {
-    type Item = String;
-    type IntoIter = IntoIter<Self::Item>;
+impl From<sqlparser::ast::Ident> for Name {
+    /// Returns a simple identifier from a sqlparser::Identifier
+    fn from(ident: sqlparser::ast::Ident) -> Self {
+        Self {
+            text: ident.value,
+            regular: ident.quote_style.is_none(),
+        }
+    }
+}
 
-    fn into_iter(self) -> Self::IntoIter {
-        self.namespace
-            .into_iter()
-            .chain(once(self.name))
-            .collect::<Vec<String>>()
-            .into_iter()
+impl From<String> for Name {
+    /// Returns a delimited simple identifier.
+    fn from(text: String) -> Self {
+        Self {
+            text,
+            regular: false,
+        }
     }
 }
 
-impl<'a> IntoIterator for &'a Identifier {
-    type Item = &'a String;
-    type IntoIter = IntoIter<Self::Item>;
+impl From<String> for Identifier {
```

Comment:
i think we should also add  `impl FromStr for Identifier`

that way you can use the build in `parse` 

```rs
let ident: Identifier = "a".parse().unwrap()
```

Reply:
actually, i noticed that there is already a `Identifier::parse` method. I'd rather convert that to `impl FromStr` instead. That gives us more flexibility as you can call `Identifier::parse("a.b")` or `"a.b".parse()`

Reply:
I've updated this to be `from_sql` and we will have a `from_path` for dot-delimited – I wanted the input domain to be explicit since there are multiple representations across the systems we are integrating with. 

I am started to lean towards removing sql identifier parsing from the python side – I don't like embedding SQL strings outside the context of queries. I've left this here for us to discuss, but I think we would only need the `from_path` for dot-delimited and could remove `from_sql`.